desktop-exporter/app/types/api-types.ts

```diff
@@ -36,6 +36,8 @@ export type SpanData = {
 
   statusCode: string;
   statusMessage: string;
+
+  depth?: number;
```

Comment:
I'm a little hesitant to extend this type with new fields given it corresponds 1:1 with OpenTelemetry's data model 🤔 Our UI information is leaking into our type definitions we use to parse API data.

If we're going to add extra data, can we add it into a specific metadata struct?

```typescript
export type SpanMetaData = {
  depth?: number
  // we will likely find other things to go here
}

export type SpanData = {
  /* ... */
  metadata?: SpanMetaData;
}
```

Alternatively, we can give `orderedSpans` a type like this, and avoid altering `SpanData`:

```typescript
type SpanWithMetadata = {
  span: SpanData;
  metadata: SpanMetaData;
}

let orderedSpans = [] as SpanWithMetadata[];
```

```typescript
orderedSpans.push({
  span: parentSpan,
  metadata: { depth }
});
```

Reply:
I went with the second solution, to avoid leaking our UI into our type definitions, especially since I expect to generate more metadata around drawing the tree that connects the spans. It took a bit of refactoring though, so I'm sorry in advance that the changes are spread across a bunch of files. 

I also moved `getTraceDurationNs` into `TraceView`, and passed it down as a prop, because we have a perfectly good array of SpanData right there, rather than separating all the spans from their metadata at `WaterfallView` level just to do this one thing. I'm not sure if this was the right call or if it matters, but that was my logic.